File: 268. Missing Number/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
  let result = 0;
  for (let i = 0; i < nums.length; i++) {
    result += (i - nums[i]);
  }
  result += nums.length;
  return result;
};

File: 268. Missing Number/solution.go
func missingNumber(nums []int) int {
	i := 0

	for i < len(nums) {
		if nums[i] < i {
			n := nums[i]
			nums[n], nums[i] = nums[i], nums[n]
		} else {
			i++
		}
	}

	for i := 0; i < len(nums); i++ {
		if i != nums[i] {
			return i
		}
	}

	return i
}